src/components/details/userDetails.js

```diff
@@ -0,0 +1,74 @@
+import { Fragment, useEffect, useState } from "react";
+import { useParams, useNavigate } from "react-router-dom";
+
+import axios from "axios";
+
+const UsersDetails = () => {
+  const navigate = useNavigate();
+  const { userId } = useParams();
+  const [details, setDetails] = useState([]);
+  const [isLoadin, setIsLoading] = useState(false);
+
+  const navigateToPostsPage = () => {
+    return navigate(`/details/${userId}/posts`);
+  };
+
+  useEffect(() => {
+    const loadUserDetails = async () => {
```

Comment:
Do not write any business login in useEffect ever, create a method for that logic and call that method in the useEffect hook